src/userCLI/src/main/kotlin/org/imperial/mrc/hint/userCLI/App.kt

```diff
@@ -36,6 +36,19 @@ fun main(args: Array<String>) {
 
     val dataSource = DbConfig().dataSource(ConfiguredAppProperties())
 
+    fun errorOutputStream(msg: String)
+    {
+        try
+        {
+            val resources = ResourceBundle.getBundle("ErrorMessageBundle", Locale("en"))
```

Comment:
Thanks for the explanation @LekanAnni , that's really useful. Still not sure why pac4j burns down the profile, but at least we can work around it! I still think it would really good to include a comment that effect in the code, just so that this issue doesn't get lost to history. Something like this, just using what you've said?
```suggestion
   // When an exception is thrown in UserLogic with key userExists, userCLI 
   // expects the key, which it then uses to get resource bundle. However,      
   //  the UserException handler for UserLogic extends HintException where   
   // additional text is appended: "HintException with key $key" so the resource
   // would not be found in the bundle.
   // After pac4j upgrade, this issue caused corruption of the profile if not dealt
   // with gracefully, hence extracting final word here. See associated ticket 
   // mrc-3549  

            val resources = ResourceBundle.getBundle("ErrorMessageBundle", Locale("en"))
```